modeling/inference_models/koboldcpp/class.py

```diff
@@ -0,0 +1,103 @@
+## GGML Backend for KoboldCpp drop in by Concedo
+## For use in KoboldAI
+
+from __future__ import annotations
+
+import time, json
+import torch
+import requests
+import numpy as np
+from typing import List, Optional, Union
+import os
+from . import koboldcpp
+
+import utils
+from logger import logger
+from modeling.inference_model import (
+    GenerationResult,
+    GenerationSettings,
+    InferenceModel,
+)
+
+model_backend_name = "koboldcpp" #specific instead of ggml
+model_backend_type = "ggml" #This should be a generic name in case multiple model backends are compatible (think Hugging Face Custom and Basic Hugging Face)
+
+kcpp_backend_loaded = False
+
+class KoboldCppException(Exception):
+    """To be used for errors on cpp side of KoboldCpp."""
+
+class KcppArgsObject:
+    def __init__(self, **kwargs):
+        self.__dict__.update(kwargs)
+
+class model_backend(InferenceModel):
+    def __init__(self) -> None:
+        super().__init__()
+
+    def is_valid(self, model_name, model_path, menu_path):
+        return "ggml" in model_name.lower()
+
+    def get_requested_parameters(self, model_name, model_path, menu_path, parameters = {}):
+        self.filename = model_name #model_path is null, name is path for some reason
+        self.model_name = "GGML_Model"
+        try:
+            from pathlib import Path
+            self.model_name = Path(model_name).name
+        except:
+            pass
+        requested_parameters = []
+        return requested_parameters
+
+    def set_input_parameters(self, parameters):
+        pass
+
+    def _load(self, save_model: bool, initial_load: bool) -> None:
+        global kcpp_backend_loaded
+        self.tokenizer = self._get_tokenizer("gpt2")
+        if not kcpp_backend_loaded:
+            kcppargs = KcppArgsObject(model=self.filename, model_param=self.filename,
```

Comment:
Since kcpp is only initialized once, is all of this config editable post-initialization?